lib/promoted/ruby/client.rb

```diff
@@ -138,7 +147,10 @@ def deliver args, headers={}
           if !@pager.validate_paging(delivery_request_builder.full_insertion, delivery_request_builder.request[:paging])
             # Invalid input, log and do SDK-side delivery.
             @logger.warn("Invalid paging parameters") if @logger
-            only_log = true
+            return {
```

Comment:
I think for the case where offset is outside the list of insertion size, it's fine to return an empty array of response insertions.

If someone modifies validate_paging to do a different check, it might not make sense to return an empty array.

Reply:
Changed, please look